server.ts

```diff
@@ -52,42 +81,117 @@ const BIRTHDATE_REGEX =
   /(^(19|20)\d\d)[-](0[1-9]|1[012])[-](0[1-9]|[12][0-9]|3[01])/;
 
 export async function setup() {
-  await createConnection();
+  const config: any = {
+    type: "postgres",
+    host: "localhost",
+    port: process.env.DB_PORT,
+    username: process.env.DB_USER,
+    password: process.env.DB_PASS,
+    database: process.env.DB_NAME,
+    synchronize: true,
+    logging: false,
+    entities: ["src/entity/**/*.ts"],
+    migrations: ["src/migration/**/*.ts"],
+    subscribers: ["src/subscriber/**/*.ts"],
+    cli: {
+      entitiesDir: "src/entity",
+      migrationsDir: "src/migration",
+      subscribersDir: "src/subscriber",
+    },
+  };
+
+  await createConnection(config);
 
   const resolvers = {
     Query: {
       hello: () => hello,
     },
     Mutation: {
-      createUser: async (_, args) => {
+      createUser: async (_, args, context) => {
         const user = new User();
         user.name = args.name;
         if (EMAIL_REGEX.test(args.email)) {
           user.email = args.email;
         } else {
-          throw new ValidationError("E-mail inválido.");
+          throw new ValidationError("E-mail inválido.", 400);
         }
-          if (PASSWORD_REGEX.test(args.password)) {
+        if (PASSWORD_REGEX.test(args.password)) {
           user.salt = bcrypt.genSaltSync(saltRounds);
           user.password = bcrypt.hashSync(args.password, user.salt);
         } else {
           throw new ValidationError(
-            "Senha deve conter no mínimo 7 caracteres com pelo menos um número e uma letra."
+            "Senha deve conter no mínimo 7 caracteres com pelo menos um número e uma letra.",
+            400
           );
         }
         if (BIRTHDATE_REGEX.test(args.birthDate)) {
           user.birthDate = args.birthDate;
         } else {
           throw new ValidationError(
-            "Data de Nascimento deve estar no formato yyyy-mm-dd"
+            "Data de Nascimento deve estar no formato yyyy-mm-dd",
+            400
+          );
+        }
+
+        return jwt.verify(context.authScope, process.env.JWT_SECRET, (err, decoded) => {
+          if (err) {
+            throw err;
+          }
+          else{
+            return getConnection().manager.save(user);
+          }
+        });
+      },
+      login: async (_, args) => {
+        if (EMAIL_REGEX.test(args.email)) {
+        } else {
+          throw new ValidationError("E-mail inválido.", 400);
+        }
```

Comment:
Aqui vc pode resumir o if pra negativa, né?


```suggestion
        if (!EMAIL_REGEX.test(args.email)) {
          throw new ValidationError("E-mail inválido.", 400);
        }
```